flashinfer/mla/_batch_mla/_contracts.py

```
@@ -0,0 +1,428 @@
+            for leaf, leaf_name in zip((left, right), split_leaf_names, strict=True):
+                if leaf.dtype != expected_dtype:
+                    raise ValueError(
+                        f"{leaf_name}.dtype={leaf.dtype} does not match the planned "
+                        f"{planned_dtype_name}={expected_dtype}."
+                    )
+        raise ValueError(f"split {name} tensor dtypes must match.")
+    if left.device != right.device:
+        raise ValueError(f"split {name} tensor devices must match.")
+    if (left.shape[-1], right.shape[-1]) != widths:
+        raise ValueError(f"split {name} last dimensions do not match planned widths.")
+    return left.dtype, tuple(left.shape[:-1]) + (sum(widths),)
```

Ruff (0.16.2)

[warning] Consider `(*tuple(left.shape[:-1]), sum(widths))` instead of concatenation

Replace with `(*tuple(left.shape[:-1]), sum(widths))`

(RUF005)